### PR TITLE
ucf101 clean

### LINKED ISSUE #791: Update UCF101 video dataset to remove artifacts
There are two issues producing artifacts:
1) ffmpeg is not called with the highest quality, producing video compression artifacts.
2) the frames are saved as jpeg, which introduces additional artifacts.

I propose modifying the TFDS file to build the dataset with high quality video extraction and keeping it saved in the original video format (not saving as frames).

### PR DESCRIPTION
**Pull request by davidslater**
Fixes #791 

This still needs to be built and use the s3 cache. Building on a small portion, it works well, though.

### CODE REVIEW

**Comment by davidslater** [comment]
Difference between frame quality:
example
example2

**Comment by davidslater** [comment]
Currently building on galileo. If successful, I will need to upload to s3 and generate checksum file.

**Comment by davidslater** [comment]
Checks out locally, and I'm in process of uploading. Otherwise, ready for review.

Can check locally by copying my cache file on galileo to your own cache directory:
`/home/davidslater/.armory/datasets/cache/ucf101_clean.tar.gz`

File: armory/data/datasets.py [hunk 1]
```diff
@@ -965,6 +966,39 @@ def ucf101(
...
+        batch_size=batch_size,
+        epochs=epochs,
+        dataset_dir=dataset_dir,
+        preprocessing_fn=preprocessing_fn,
+        as_supervised=False,
+        supervised_xy_keys=("video", "label"),
+        variable_length=bool(batch_size > 1),
+        cache_dataset=cache_dataset,
+        framework=framework,
+        shuffle_files=shuffle_files,
```
Comment (ng390) [inline armory/data/datasets.py:998]: With the (very) recently merged in PR #928, _generator_from_tfds now takes a context to facilitate the export process.

Comment (davidslater) [reply armory/data/datasets.py:998]: Should be fixed now

**Review by ng390 — APPROVED** [review]
LGTM

**Comment by ng390** [comment]
> LGTM

Upload completed.

**Pull request merged**